warnings building echo_client with nrf5

`make BOARD=nrf52840_pca10056 CONF_FILE=prj_nrf5.conf`

results in:

```
 CC      drivers/gpio/gpio_nrf5.o
  LD      drivers/gpio/built-in.o
  CC      drivers/ieee802154/ieee802154_nrf5.o
In file included from /home/nashif/Work/github/zephyr/include/net/net_if.h:24:0,
                 from /home/nashif/Work/github/zephyr/drivers/ieee802154/ieee802154_nrf5.c:21:
/home/nashif/Work/github/zephyr/drivers/ieee802154/ieee802154_nrf5.c:449:16: warning: dereferencing type-punned pointer will break strict-aliasing rules [-Wstrict-aliasing]
       ((struct nrf5_802154_data *)
                ^
/home/nashif/Work/github/zephyr/include/net/net_core.h:113:12: note: in definition of macro ‘NET_STACK_INFO_ADDR’
   .stack = _addr,      \
            ^~~~~
  LD      drivers/ieee802154/built-in.o
```

--- a/drivers/ieee802154/ieee802154_nrf5.c
+++ b/drivers/ieee802154/ieee802154_nrf5.c
@@ -446,6 +446,5 @@ NET_DEVICE_INIT(nrf5_154_radio, CONFIG_IEEE802154_NRF5_DRV_NAME,
 NET_STACK_INFO_ADDR(RX, nrf5_154_radio,
 		    CONFIG_IEEE802154_NRF5_RX_STACK_SIZE,
 		    CONFIG_IEEE802154_NRF5_RX_STACK_SIZE,
-		    ((struct nrf5_802154_data *)
-		    (&__device_nrf5_154_radio))->rx_stack, 0);
+		    nrf5_data.rx_stack, 0);
 #endif

